### ISSUE TITLE
NavigateToString doesn't load CSS files and images

### ISSUE THREAD

**Issue by michaldivis** [MicrosoftEdge/WebView2Feedback#692]
**Description**
I'm trying to use the `NavigateToString` method do load HTML code, however, the CSS file doesn't get used and the image isn't displayed.

**Version**
SDK: 1.0.664.37
Runtime: 87.0.664.47
Framework: WPF (.NET 5)
OS: Win10 2004 (build 19041.572)

**Repro Steps**

My app looks like this...

MainWindow.xaml
```xaml
<Window
    x:Class="WpfUI.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:wv2="clr-namespace:Microsoft.Web.WebView2.Wpf;assembly=Microsoft.Web.WebView2.Wpf"
    Title="MainWindow"
    Width="800"
    Height="450"
    WindowStartupLocation="CenterScreen"
    mc:Ignorable="d">
    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="auto" />
            <RowDefinition Height="*" />
        </Grid.RowDefinitions>
        <WrapPanel Grid.Row="0">
            <Button
                Margin="10,10,0,10"
                Padding="5"
                Click="BtnPageOne_Click"
                Content="Page One" />
        </WrapPanel>
        <wv2:WebView2
            Name="webView"
            Grid.Row="1"
            CoreWebView2Ready="webView_CoreWebView2Ready" />
    </Grid>
</Window>
```

MainWindow.xaml.cs
```csharp
 public partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();
    }

    protected override async void OnContentRendered(EventArgs e)
    {
        base.OnContentRendered(e);
        await webView.EnsureCoreWebView2Async();
    }

    private void BtnPageOne_Click(object sender, RoutedEventArgs e)
    {
        OpenStringHtml(@"C:\Users\Me\Desktop\test.html");
    }

    private void OpenStringHtml(string htmlFilePath)
    {
        var html = File.ReadAllText(htmlFilePath);
        webView.NavigateToString(html);
    }
}
```

test.html
```html
<html>

<head>
    <meta charset="utf-8">
    <meta name="viewport"
        content="width=device-width, initial-scale=1.0, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <link rel="stylesheet" href="C:\Resources\bootstrap.min.css">
</head>

<body>
    <div class="container-fluid">
        <p>Some text</p>
        <image src="C:\Users\Me\Desktop\car.jpg" width="120">
    </div>
</body>

</html>
```

***Expected behavior***
The CSS style gets used and the image gets displayed

***Actual behavior***
The CSS style doesn't get used and the image isn't displayed

**Comment by champnic** [comment]
There's currently an issue with loading files from NavigateToString code related to CORS. We are tracking a fix in #530. As a workaround, you should be able to navigate directly to your local file using the "file:///" protocol. You may also want to try out the SetVirtualHostNameToFolderMapping function (1.0.707-prerelease + Dev/Canary Edge required as it's still experimental):
https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.setvirtualhostnametofoldermapping?view=WebView2-dotnet-1.0.707-prerelease#Microsoft_Web_WebView2_Core_CoreWebView2_SetVirtualHostNameToFolderMapping_System_String_System_String_Microsoft_Web_WebView2_Core_CoreWebView2HostResourceAccessKind_

**Comment by cemocfr** [comment]
Hi, we have a similar issue but the css file is handled by a custom protocol, so it's not possible for us to us the proposed workarround nor the new SetVirtualHostNameToFolderMapping (the file is not physically on disk).
For our use case it would be great if you could notify the webresourcerequested event when NavigateToString is used as we could handle our protocol in the handler.
